File: Kintino.CipherConf.Crypto/Implementations/NonceGenerator.cs
using Kintino.CipherConf.Crypto.Services.Algos;
using Kintino.CipherConf.Primitives;

namespace Kintino.CipherConf.Crypto.Implementations;

internal class NonceGenerator : INonceGenerator
{
    Nonce INonceGenerator.NewNonce() => new(new PlainData(RandomNonceGenerator.GenerateNonce()));
}

File: EncryptConfig.Js/src/modules/encryption/src/services/NonceGenerator.ts
import type { INonceGenerator, Nonce } from "@abstractions";
import { Rng } from "./Rng";

const nonceSizeInBytes = 12; // 96 bits

export class NonceGenerator implements INonceGenerator {
	newNonce(): Nonce {
		return Rng.createRandomBytes(nonceSizeInBytes); // 96 bits
	}
}
